app-android/src/main/java/com/ustadmobile/port/android/view/composable/UstadCourseBlockEdit.kt

```diff
@@ -0,0 +1,134 @@
+package com.ustadmobile.port.android.view.composable
+
+import androidx.compose.foundation.layout.*
+import androidx.compose.material.*
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.toughra.ustadmobile.R
+import com.ustadmobile.core.impl.locale.entityconstants.CompletionCriteriaConstants
+import com.ustadmobile.core.viewmodel.CourseBlockEditUiState
+import com.ustadmobile.lib.db.entities.CourseBlock
+import com.ustadmobile.lib.db.entities.ext.shallowCopy
+
+@Composable
+fun UstadCourseBlockEdit(
+    uiState: CourseBlockEditUiState,
+    onCourseBlockChange: (CourseBlock?) -> Unit = {}
+){
+
+    Column{
+
+        UstadDateTimeEditTextField(
+            value = uiState.courseBlock?.cbHideUntilDate ?: 0,
+            dateLabel = stringResource(id = R.string.dont_show_before)
+                .addOptionalSuffix(),
+            timeLabel = stringResource(id = R.string.time),
+            timeZoneId = uiState.timeZone
+        )
+
+        Text(stringResource(R.string.class_timezone_set, uiState.timeZone))
+
+        Spacer(modifier = Modifier.height(10.dp))
+
+        Row() {
+            UstadMessageIdOptionExposedDropDownMenuField(
+                modifier = Modifier.weight(0.5F),
+                value = uiState.courseBlock?.cbCompletionCriteria ?: 0,
+                label = stringResource(R.string.completion_criteria),
+                options = CompletionCriteriaConstants.COMPLETION_CRITERIA_MESSAGE_IDS,
+                onOptionSelected = {
+                    onCourseBlockChange(uiState.courseBlock?.shallowCopy{
+                        cbCompletionCriteria = it.value
+                    })
+                },
+                enabled = uiState.fieldsEnabled,
+            )
+
+            Spacer(modifier = Modifier.width(15.dp))
+
+            if (uiState.minScoreVisible){
```

Comment:
This will crash:

The value of the text will be something like "5 points". If you try "5 points".toInt() - it will crash because the string is not valid for parsing into an int.

The "points" string or percentage etc needs to added as an adornment (like a trailing icon), not in the value of the string itself. I have added a property called suffix to UstadTextEditField on React and Jetpack compose that will do this. You can use this for the points string and the percentage string. I have also included examples in the previews.

We also need to set the keyboard type to numeric so that users will see only the numbers on the keyboard, not all the letters. I have added KeyboardOptions as a parameter on UstadTextEditField, which is then passed to the textfield we are using. This is also included in the example. There is an example also for setting the inputType on React (this will have an effect if the user opens the web version using a mobile browser).

Finally, we need to filter out anything the user might try to type that is not a number to be completely sure that we won't have invalid input. On Android it is still possible to connect a bluetooth keyboard, or copy/paste something invalid. We cannot rely only on showing the numerical keyboard.

For that we need to adjust the onChange function to use something like this:

```
onChange = { newString ->
   uiState.entity.numericalProperty = newString.filter { it.isDigit() }.toInt()
}
```